src/components/PurchaseOrderItem.vue

```diff
@@ -26,8 +26,23 @@ export default defineComponent({
   },
   props: ["purchaseOrder"],
   methods: {
-    async viewPurchaseOrder () {
-      this.router.push({ path: `/purchase-order/${this.purchaseOrder.orderId}` })
+    async getOrderDetail(orderId?: any) {
+      const payload = {
```

Comment:
I think we should decide in action whether to make an API call or not. We should pass orderId, and in action we decide whether we want extract from existing list or make API call